### PR TITLE
adding internal tracetoken and trigger source

### PR DESCRIPTION
**Pull request by laurence-bird**

### CODE REVIEW

File: src/main/scala/com/ovoenergy/orchestration/Main.scala [hunk 1]
```diff
@@ -75,7 +75,8 @@ object Main extends App
       hosts = config.getString("kafka.hosts"),
       groupId = config.getString("kafka.group.id"),
       topic = config.getString("kafka.topics.triggered")
-    )
+    ),
+    () => "HI"
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/orchestration/Main.scala]: what's this?

File: src/test/scala/com/ovoenergy/orchestration/processes/FailureSpec.scala [hunk 2]
```diff
@@ -2,18 +2,23 @@ package com.ovoenergy.orchestration.processes
...
 import org.scalatest.{FlatSpec, Matchers}
+import org.scalacheck.Shapeless._
 
 import scala.concurrent.Future
 
 class FailureSpec extends FlatSpec
   with Matchers
   with ScalaFutures {
 
+  private def generate[A](a: Arbitrary[A]) = {
```
Comment (cb372) [inline src/test/scala/com/ovoenergy/orchestration/processes/FailureSpec.scala]: You could write this as

```scala
private def generate[A: Arbitrary] = implicitly[Arbitrary[A]].sample.get
```

and thus call it more simply:

```scala
val triggered = generate[Triggered]
```

Comment (laurence-bird) [inline src/test/scala/com/ovoenergy/orchestration/processes/FailureSpec.scala]: ahh yes, nice. will update cheers

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 66](https://circleci.com/gh/ovotech/comms-orchestration/66) at 2017-01-12 17:38 GMT.

**Pull request merged**